### PR TITLE
monday api integration

### PR DESCRIPTION
**Pull request by saudkhan214**
This PR includes Monday.com API integration into elb using OAuth flow. 
In this PR the issues# 5 - 7 from Board have been completed.
## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests to cover the new feature or fixes.

### CODE REVIEW

**Review by elbalexandre — COMMENTED** [review]
Great work !

Could you try to re-enable edge runtime configuration everywhere, I don't see any reason it could not work (it's way cheaper for us):
- reset each endpoint (including inngest one) with default template config (`preferredRegion` + `runtime`)
- reset `docker-compose` as well

File: apps/monday/src/app/auth/callback/service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  });
+
+  await inngest.send({
+    name: 'monday/users.page_sync.requested',
+    data: {
+      isFirstSync: true,
+      organisationId,
+      region,
+      syncStartedAt: Date.now(),
+      page: 1,
```
[the remark this answers is not in the document]
Comment (saudkhan214) [reply apps/monday/src/app/auth/callback/service.ts:35]: According to doc it starts from 1
image

FYI: https://developer.monday.com/api-reference/reference/users#arguments

File: apps/monday/src/connectors/users.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    });
+
+    // test('should return users and no nextPage when the token is valid and their is no other page', async () => {
+    //   await expect(getUsers(validToken, maxPage)).resolves.toStrictEqual({
+    //     data:{
+    //       users
+    //     },
+    //     nextPage: null,
+    //   });
+    // });
```
[the remark this answers is not in the document]
Comment (saudkhan214) [reply apps/monday/src/connectors/users.test.ts:73]: I tried this test but it fails, reason is the API it self does not give us the nextPage, We have to determine it by ourself by making additional request with page increment. So suppose if we make a request with nextPage=1 and it gives us the data then next time we make another request with nextPage=2, but this time it does not return any data so we make nextPage=null to stop the iteration, but the Test is supposed to return the data with nextPage=null

File: apps/monday/src/app/auth/callback/service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (elbalexandre) [reply apps/monday/src/app/auth/callback/service.ts:35]: Weird

File: apps/monday/src/connectors/users.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (elbalexandre) [reply apps/monday/src/connectors/users.test.ts:73]: So could you add a test that checking that the users array is empty + nextPage = null when the page index is out of bound ? (you can remove the commented test)

File: packages/test-utils/src/inngest/function-mock.ts [hunk 3]
```diff
@@ -76,3 +76,34 @@ export const createInngestFunctionMock =
       context,
     ] as MockSetupReturns<F, EventName>;
   };
+
+  
+export const createFunctionMock =
```
Comment (elbalexandre) [inline packages/test-utils/src/inngest/function-mock.ts:81]: Can you revert theses changes ? It's unused.

Comment (saudkhan214) [reply packages/test-utils/src/inngest/function-mock.ts:81]: I'm using it.
image

Comment (elbalexandre) [reply packages/test-utils/src/inngest/function-mock.ts:81]: just use `createInngestFunctionMock`, if any feature are missing you can edit it.

**Comment by elbalexandre** [comment]
Everything looks right, but please:
- use `createInngestFunctionMock`
- run `npx prettier --fix`
- resolve conflict as well (`pnpm i`)

**Review by elbalexandre — APPROVED** [review]
